tests/integrations/gateway/test_gateway.py

```diff
@@ -0,0 +1,53 @@
+# pylint: disable=unused-import,redefined-outer-name # fixture
+import requests
+from scaleway.container.v1beta1 import Container
+
+from tests import constants
+from tests.app_fixtures.routed_functions import MESSAGES
+from tests.integrations.deploy_wrapper import run_deploy_command
+from tests.integrations.gateway_fixtures import api_gateway, auth_key  # noqa
+from tests.integrations.project_fixture import scaleway_project  # noqa
+from tests.integrations.utils import create_client
+
+
+def test_integration_gateway(
+    scaleway_project: str, api_gateway: Container, auth_key: str  # noqa
+):
+    client = create_client()
+    client.default_project_id = scaleway_project
+
+    gateway_url = f"https://{api_gateway.domain_name}"
+
+    run_deploy_command(
+        client,
+        constants.APP_FIXTURES_PATH / "routed_functions.py",
+        "--gateway-url",
+        gateway_url,
+        "--gateway-api-key",
+        auth_key,
+    )
+
+    # Check general routing configuration
+    resp = requests.get(
+        url=gateway_url + "/health", timeout=constants.COLD_START_TIMEOUT
+    )
+    assert resp.status_code == 200
+    assert resp.text == MESSAGES["/health"]
+
+    # Test with common prefix with configured routes
+    resp = requests.get(
+        url=gateway_url + "/messages", timeout=constants.COLD_START_TIMEOUT
+    )
+    assert resp.status_code == 200
+    assert resp.text == MESSAGES["/messages"]
+
+    # Check a route with a method that is not configured
+    resp = requests.get(
+        url=gateway_url + "/messages/new", timeout=constants.COLD_START_TIMEOUT
+    )
+    assert resp.status_code == 404
```

Comment:
Waiting on the changes to the gateway to be merged for this to work :safety_vest: